portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,27 +28,72 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.io.IOUtils;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private List<String> comments;
-    @Override
-    public void init(){
-        comments = new ArrayList<>();
-        comments.add("This is the first comment");
-        comments.add("This is the second comment");
-        comments.add("This is the third comment");
+    public class CommentClass{
+        String comment;
+        String userName;
+        long timestamp;
+
+        public CommentClass(String initComment, String initUserName, long initTimestamp){
+            comment = initComment;
+            userName = initUserName;
+            timestamp = initTimestamp;
+        }
     }
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
you can chain this with the prepare call in the next line because you don't reuse the datastore variable elsewhere in the function.

Reply:
Agreed. Resolved